platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/EmailServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.email.impl;
+
+import static org.apache.commons.lang3.Validate.inclusiveBetween;
+import static org.apache.commons.lang3.Validate.notEmpty;
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.Properties;
+
+import javax.mail.Authenticator;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+
+import org.apache.commons.lang3.StringUtils;
+import org.codice.ddf.platform.email.api.EmailService;
+
+/**
+ * Supports unauthenticated connections to a mail server and username/password authenticated connections
+ * over TLS. If the username is not blank and the password is not empty, then auth/tls will be used.
+ */
+public class EmailServiceImpl implements EmailService {
+
+    private static final String SMTP_HOST_PROPERTY = "mail.smtp.host";
+
+    private static final String SMTP_PORT_PROPERTY = "mail.smtp.port";
+
+    private static final String SMTP_AUTH_PROPERTY = "mail.smtp.auth";
+
+    private static final String SMTP_START_TLS_ENABLE_PROPERTY = "mail.smtp.starttls.enable";
+
+    private static final String TRUE = "true";
+
+    private static final String FALSE = "false";
+
+    private String hostName;
+
+    private Integer portNumber;
+
+    private String userName;
+
+    private String password;
+
+    /**
+     * Set the username for the email server.
+     *
+     * @param userName may be null or blank
+     */
+    public void setUserName(String userName) {
+        this.userName = userName;
+    }
+
+    /**
+     * Set the password for the email server.
+     *
+     * @param password may be null or blank
+     */
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    /**
+     * @param hostName must be non-empty
+     */
+    public void setHostName(String hostName) {
+        notEmpty(hostName, "hostName must be non-empty");
+        this.hostName = hostName;
+    }
+
+    /**
+     * @param portNumber must be non-null and >0
+     */
+    public void setPortNumber(Integer portNumber) {
+        notNull(portNumber, "portNumber must be non-null");
+        inclusiveBetween(1, Integer.MAX_VALUE, portNumber, "portNumber must be >0");
+        this.portNumber = portNumber;
+    }
+
+    @Override
+    public Session createSession() {
+
+        Properties properties = System.getProperties();
+
+        properties.setProperty(SMTP_HOST_PROPERTY, hostName);
+        properties.setProperty(SMTP_PORT_PROPERTY, portNumber.toString());
+
+        if (StringUtils.isNotBlank(userName) && StringUtils.isNotEmpty(password)) {
+            properties.put(SMTP_AUTH_PROPERTY, TRUE);
+            properties.put(SMTP_START_TLS_ENABLE_PROPERTY, TRUE);
+
+            return Session.getInstance(properties, new Authenticator() {
```

Comment:
So, If a bad hostname is entered, an error appears when the email is about to be sent. Example:

10:42:00,296 | WARN  | heduler_Worker-2 | log.ui.query.monitor.email.EmailNotifier  156 | catalog-ui-search    | unable to send email to admin@localhost.local
com.sun.mail.util.MailConnectException: Couldn't connect to host, port: connexta-com.mail.protection.outlook.co, 25; timeout -1;
  nested exception is:
	java.net.UnknownHostException: connexta-com.mail.protection.outlook.co

The error message includes the hostname, so an admin should be able to diagnose the problem with default log settings.